testng-core-api/src/main/java/org/testng/internal/Utils.java

```diff
@@ -333,7 +333,13 @@ public static String shortStackTrace(Throwable t, boolean toHtml) {
   private static String buildStackTrace(Throwable t, boolean toHtml, StackTraceType type) {
     StringWriter sw = new StringWriter();
     PrintWriter pw = new PrintWriter(sw);
-    t.printStackTrace(pw);
+    try {
+      t.printStackTrace(pw);
+    } catch (Throwable e) {
+      // failsafe in case if printStackTrace throws exception
+      pw.println(t.getClass().getName());
```

Comment:
this may be not perfect exception handling, but in case if `t.printStackTrace(pw)` throws exception (see #2807) we do not have much options here